### PR TITLE
[UNOMI-853] Adapt migration job to use asynchronous mode avoiding timeout and connection lost

### PR DESCRIPTION
**Pull request by jayblanc**
[UNOMI-853](https://issues.apache.org/jira/browse/UNOMI-853) **Adapt migration job to use asynchronous mode avoiding timeout and connection lost**

Reindex step during migration has been moved to a async task and check against task completion is done until task finish.  
Update a link to karaf in the doc that was broken.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

### CODE REVIEW

File: tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java [hunk 1]
```diff
@@ -216,7 +216,9 @@ public static void reIndex(CloseableHttpClient httpClient, BundleContext bundleC
             // Recreate the original index with new mappings
             HttpUtils.executePutRequest(httpClient, esAddress + "/" + indexName, newIndexSettings, null);
             // Reindex data from clone
-            HttpUtils.executePostRequest(httpClient, esAddress + "/_reindex", reIndexRequest, null);
+            JSONObject task = new JSONObject(HttpUtils.executePostRequest(httpClient, esAddress + "/_reindex?wait_for_completion=false", reIndexRequest, null));
+            //Wait for the reindex task to finish
+            waitForTaskToFinish(httpClient, esAddress, task.getString("task"), migrationContext);
```
Comment (jsinovassin) [inline tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java:221]: A reindex query is also called in the method MigrationUtils.moveToIndex, this one might also need to be modified like it's done here

File: tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java [hunk 2]
```diff
@@ -281,6 +283,23 @@ public static void waitForYellowStatus(CloseableHttpClient httpClient, String es
...
+                            null));
+            if (status.has("completed") && status.getBoolean("completed")) {
+                migrationContext.printMessage("Task is completed");
+                break;
+            }
+            if (status.has("error")) {
+                final JSONObject error = status.getJSONObject("error");
+                throw new IOException("Task error: " + error.getString("type") + " - " + error.getString("reason"));
+            }
+            migrationContext.printMessage("Waiting for Task " + taskId + " to complete");
```
Comment (jsinovassin) [inline tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java:299]: Don't we have a risk to have too much logs here. 
According to the time of the reindex, we might have too much log displayed.

**Review by jsinovassin — CHANGES_REQUESTED** [review]

File: tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java (hunk 2 above)
Comment (jayblanc) [reply tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java:299]: Yes I know but migration is not something regular, and knowing that something is running may be useful. What should be the frequency for that log ? 5mn ?

File: tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java (hunk 1 above)
Comment (jayblanc) [reply tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java:221]: Fixed

File: tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java (hunk 2 above)
Comment (jsinovassin) [reply tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/utils/MigrationUtils.java]: Every 20-30 secondes is a good option I think

**Review by jsinovassin — APPROVED** [review]

**Pull request merged**